kmir/src/kmir/kdist/mir-semantics/alloc.k

```diff
@@ -12,15 +15,28 @@ module ALLOC
   imports INT
 
 syntax BinderForExistentialTraitRef ::= binderForExistentialTraitRef(value: ExistentialTraitRef, boundVars: BoundVariableKindList)
-syntax MaybeBinderForExistentialTraitRef ::= someBinderForExistentialTraitRef(BinderForExistentialTraitRef)
-                                           | "noBinderForExistentialTraitRef"
-syntax GlobalAlloc ::= globalAllocFunction(Instance)
-                     | globalAllocVTable(Ty, MaybeBinderForExistentialTraitRef)
-                     | globalAllocStatic(StaticDef) [symbol(globalAllocStatic)]
-                     | globalAllocMemory(Allocation) [symbol(globalAllocMemory)]
-
-syntax GlobalAllocEntry ::= globalAllocEntry(MIRInt, GlobalAlloc) [symbol(globalAllocEntry)]
-syntax GlobalAllocsMap ::= List {GlobalAllocEntry, ""} [symbol(globalAllocsMap), terminator-symbol(.globalAllocsMap)]
+                                          [group(mir---value--bound-vars)]
+
+syntax MaybeBinderForExistentialTraitRef ::= someBinderForExistentialTraitRef(BinderForExistentialTraitRef) [group(mir-option)]
+                                           | "noBinderForExistentialTraitRef"                               [group(mir-option)]
+
+syntax GlobalAllocKind ::= globalAllocFunction(Instance)
+                             [group(mir-enum), symbol(GlobalAllocKind::Function)]
+                         | globalAllocVTable(Ty, MaybeBinderForExistentialTraitRef)
+                             [group(mir-enum), symbol(GlobalAllocKind::VTable)]
+                         | Static(StaticDef)
+                             [group(mir-enum), symbol(GlobalAllocKind::Static)]
+                         | Memory(Allocation)
+                             [group(mir-enum), symbol(GlobalAllocKind::Memory)]
```

Comment:
I think the reason that this was called `GlobalAlloc` and not `GlobalAllocKind` is that it is specifically referencing [GlobalAlloc in the Stable MIR AST](https://github.com/runtimeverification/rust/blob/85f90a461262f7ca37a6e629933d455fa9c3ee48/compiler/stable_mir/src/mir/alloc.rs#L8-L22). I think the intention was to keep the non-terminals sorts the same, but I don't know if we have already deviated or if we ought to.

Reply:
Yes, that was the case.

Reply:
Well... but we have the convention for `mir-list` that the list sort is the plural of the non-list sort.
Maybe we can just use untyped K-lists in the semantics, though, then we can rename things freely.

Reply:
That's true!

In any case, all these names were given as they are in the Stabe MIR AST, and then any additional structures that were created were made to not clash.
Then, when the parser was written, we created a collection of guidelines that needed to be followed so that we could identify which rule we need to apply. If there is the need to change names to achieve that, this would take priority.

I am sure that it is possible to change the names of the sorts that we created that were not directly in the Stable MIR AST in order to keep the ones that are the same, if this is something that you think is important. The reasoning behind doing it this way initially was to easily know which sort corresponds to which part of the AST.